src/com/lftechnology/java/training/krishna/employeemanagement/dao/employee/impl/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,291 @@
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.lftechnology.java.training.krishna.employeemanagement.dao.employee.EmployeeDao;
+import com.lftechnology.java.training.krishna.employeemanagement.domain.Employee;
+import com.lftechnology.java.training.krishna.employeemanagement.jdbc.DbConnectionFactory;
+import com.lftechnology.java.training.krishna.employeemanagement.utils.ConstantUtils;
+
+/**
+ * Implementation of employee DAO interface.
+ *
+ * @author Krishna Timilsina <krishnatimilsina@lftechnology.com>
+ * @version 1.0
+ */
+public class EmployeeDaoImpl implements EmployeeDao {
+
+	private static final Logger LOGGER =
+		Logger.getLogger(EmployeeDaoImpl.class.getName());
+
+	Connection connection = DbConnectionFactory.getDbConnection();
+	PreparedStatement preparedStatement = null;
+	ResultSet rs = null;
+	private static final String INSERT_USER_QUERY =
+		"INSERT INTO user (username, password, fullname, department, address,role, isTerminated, createdAt) VALUES (?, ?, ?, ?, ?, ?, ?, ?)";
+	private static final String SEARCH_USER_QUERY =
+		"SELECT * FROM user WHERE fullname LIKE ? OR department LIKE ? OR address LIKE ?";
+	private static final String DELETE_USER_QUERY =
+		"DELETE FROM user WHERE fullname = ?";
+	private static final String TERMINATE_USER_QUERY =
+		"UPDATE user SET isTerminated = ?, modifiedAt= ?  WHERE fullname = ?";
+	private static final String SEARCH_BY_ID_QUERY =
+		"SELECT * FROM user WHERE id = ?";
+	private static final String SEARCH_BY_USERNAME_QUERY =
+		"SELECT * FROM user WHERE username = ?";
+	
+	
+
+	public EmployeeDaoImpl() {
+
+	}
+
+	@Override
+	public Employee create(Employee entity) {
+
+		try {
+			preparedStatement =
+				connection.prepareStatement(
+					INSERT_USER_QUERY, Statement.RETURN_GENERATED_KEYS);
+			preparedStatement.setString(1, entity.getUsername());
+			preparedStatement.setString(2, entity.getPassword());
+			preparedStatement.setString(3, entity.getFullname());
+			preparedStatement.setString(4, entity.getDepartment());
+			preparedStatement.setString(5, entity.getAddress());
+			preparedStatement.setString(6, entity.getRole());
+			preparedStatement.setInt(7, entity.getIsTerminated());
+			preparedStatement.setDate(8, entity.getCreatedAt());
+			int affectedRows = preparedStatement.executeUpdate();
+			if (affectedRows == 0) {
+				throw new SQLException(
+					"Creating user failed, no rows affected.");
+			}
+			ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+			if (generatedKeys.next()) {
+				entity.setId(generatedKeys.getInt(1));
+			}
+		}
+		catch (SQLException e) {
+			LOGGER.log(
+				Level.SEVERE, "Exception while creating statement: {0}", e);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Exception while creating statement: {0}" 7 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)